src/storage/blockchain/onchain-issuer-adapter/non-merklized/version/v0.0.1/onchain-non-merklized-issuer-adapter.ts

```diff
@@ -0,0 +1,309 @@
+          break;
+        }
+        case XSDNS.String: {
+          this.validateSourceValue(dataType, f.value, f.rawValue);
+          credentialSubject[f.key] = f.rawValue;
+          break;
+        }
+        case XSDNS.DateTime: {
+          const timestamp = BigInt(f.rawValue);
+          const sourceTimestamp = getDateFromUnixTimestamp(Number(timestamp)).toISOString();
+          this.validateSourceValue(dataType, f.value, sourceTimestamp);
+          credentialSubject[f.key] = sourceTimestamp;
+          break;
+        }
+        case XSDNS.Double: {
+          const rawFloat = Number(f.rawValue);
+          this.validateSourceValue(dataType, f.value, rawFloat);
+          credentialSubject[f.key] = rawFloat;
+          break;
+        }
+        default: {
+          throw new Error(`Unsupported data type ${dataType}`);
+        }
+      }
+    }
+    credentialSubject['type'] = credentialType;
+
+    const subjectId = coreClaim.getId();
+    const subjectDid = DID.parseFromId(subjectId);
+    credentialSubject['id'] = subjectDid.string();
+
+    return credentialSubject;
+  }
+
+  private async existenceProof(coreClaim: Claim): Promise<Iden3SparseMerkleTreeProof> {
+    const [mtpProof, stateInfo] = await this._contract.getClaimProofWithStateInfo(
+      coreClaim.hIndex()
+    );
+    if (!mtpProof.existence) {
+      throw new Error('Claim does not exist');
+    }
+    const latestStateHash = Hash.fromBigInt(stateInfo.state);
+    const latestClaimsOfRootHash = Hash.fromBigInt(stateInfo.claimsRoot);
+    const latestRevocationOfRootHash = Hash.fromBigInt(stateInfo.revocationsRoot);
+    const latestRootsOfRootHash = Hash.fromBigInt(stateInfo.rootsRoot);
+
+    const p = new Proof({
+      siblings: mtpProof.siblings.map((s) => Hash.fromBigInt(s)),
+      existence: mtpProof.existence,
+      nodeAux: mtpProof.auxExistence
+        ? {
+            key: Hash.fromBigInt(mtpProof.auxIndex),
+            value: Hash.fromBigInt(mtpProof.auxValue)
+          }
+        : undefined
+    });
+
+    return new Iden3SparseMerkleTreeProof({
+      issuerData: {
+        id: this._issuerDid,
+        state: {
+          value: latestStateHash,
+          claimsTreeRoot: latestClaimsOfRootHash,
+          revocationTreeRoot: latestRevocationOfRootHash,
+          rootOfRoots: latestRootsOfRootHash
+        }
+      },
+      mtp: p,
+      coreClaim: coreClaim
+    });
+  }
+
+  private async validateSourceValue(dataType: string, originHash: bigint, source: unknown) {
+    const sourceHash = await Merklizer.hashValue(dataType, source);
+    if (sourceHash !== originHash) {
+      throw new Error(`Invalid source value for ${dataType} type`);
+    }
+  }
+
+  private convertDisplayMethod(
+    onchainDisplayMethod: INonMerklizedIssuer.DisplayMethodStructOutput
+  ): DisplayMethod | undefined {
+    if (onchainDisplayMethod.id === '' && onchainDisplayMethod._type === '') {
+      return undefined;
+    }
```

Comment:
```suggestion
    if (!onchainDisplayMethod.id && !onchainDisplayMethod._type) {
      return null;
    }
```

Reply:
Can I use `undefined` here? Because I use the convertor method to convert displayMethod and then use it for W3cCredential, where this field is DisplayMethod | undefined.

Reply:
Did changes for if